### ISSUE TITLE
`create` with generic driver and two factor authentication

### ISSUE THREAD

**Issue by datacarl** [docker/machine#3175]
There doesn't seem to be a way to provision a host that requires two factor authentication for SSH. Machine stops at waiting for SSH to be available. 

```
docker-machine create --driver "generic" --swarm --swarm-master --generic-ip-address $IP --generic-ssh-user $USER --generic-ssh-key $SSH_KEY swarm-master-aws
Running pre-create checks...
Creating machine...
(swarm-master-aws) Importing SSH key...
Waiting for machine to be running, this may take a few minutes...
Detecting operating system of created instance...
Waiting for SSH to be available...
Error creating machine: Error detecting OS: Too many retries waiting for SSH to be available.  Last error: Maximum number of retries (60) exceeded
```

As two factor auth becomes more widely used this seems like a feature that would be greatly beneficial to a lot of users. Docker Machine does such a great job provisioning machines that it would be a shame having to do it manually whenever the server requires two factor auth. Especially not having to copy certificates back and forth and configure swarm is quite a relief.

**Comment by nathanleclaire** [comment]
When you say "two-factor authentication for SSH", what do you mean exactly?  Password-based SSH keys?  Usually when I hear 2FA I think of having to type in a code in that's delivered to you via SMS or other means, I'm assuming you mean a different kind here.

**Comment by datacarl** [comment]
Hey @nathanleclaire, thanks for getting back!

> Usually when I hear 2FA I think of having to type in a code in that's delivered to you via SMS or other means

Yes that's the one. On top of your ssh key pair you would also auth with a one time code. I've encountered this a couple of times on servers in security aware organisations. I do not have a any numbers on how common it is but [googling](https://www.google.com/search?q=2FA%20linux&rct=j) should give an idea of what's out there. There are both commercial and free options available. 

Right now 2FA makes `docker-machine` incapable of installing anything.

**Comment by nathanleclaire** [comment]
How does that work usually (outside of Machine)?  You run `ssh` (OpenSSH?) and it will prompt you for a code, delivered to you by SMS?

**Comment by datacarl** [comment]
Yeah, the first factor is usually your key pair (which you provide as usual). When that is verified you will be prompted for the second factor (sms/code from authenticator app/yubikey). In theory you could have more factors, like a static password that you'd also be prompted for, but I'd say that's probably pretty rare.

**Comment by sebadiaz** [comment]
#3509

**Comment by stormfish-sci** [comment]
Take a look at https://duo.com/docs/duounix for an example 2FA implementation that precludes docker-machine from working.
